routes/books.js

```diff
@@ -27,16 +27,17 @@ router.route('/')
     ])
   })
   .post(function(req, res) {
-    res.statusCode = 200;
-    res.json({
-      "id": 127,
-      "user_id": 1,
-      "title": "Test Book 3: The Bookening",
-      "author": "An Author Name",
-      "isbn": "1234567890123",
-      "summary": "An Author Name concludes his thrilling horror series with a new age children's book",
-      "visibility": true
-    })
+    let book = req.body;
+    db.books.insert(book).then((result) => {
+      if (result[0].affectedRows === 0) {    // If no rows are affected, send a 404.
+        res.statusCode = 404;
+        res.end();
+        return
+      }
+       res.statusCode = 200;
+       res.send(result.insertId);
+       res.end();
```

Comment:
I would use `res.json({ "id": result.insertId });` here instead of `res.send()` and `res.end()`